### ISSUE TITLE
A simple example to showcase autodiff

### ISSUE THREAD

**Issue by chaithyagr** [mind-inria/mri-nufft#104]
We could have possibly the following examples:
- [ ] A simple U-Net (pulled from fastMRI dataset possibly), trained? @Lenoush can you handle this in your free time?
- [ ] A network built with DeepInv library for some already preset reconstruction (PnP possibly?) @matthieutrs can you handle this this in your free time? I think you can also have something basic passed to @Lenoush and she can work on it forward.
- [x] A simple example where we could learn the trajectory possibly? @alineyyy can you handle this in your free time? #167

**Comment by paquiteau** [comment]
Beware, there is also autodiff for the image/k-space data (the one that is only currently supported) This could also be show cased.

**Comment by chaithyagr** [comment]
I really seemed to have clicked open issue too soon. I had written so much more xD. Lemme update

**Comment by chaithyagr** [comment]
@AsmaTANABEN , can you have a look at the notebook shared in mattermost and see if we can get this on mri-nufft?

**Comment by chaithyagr** [comment]
Reopening as it got closed by mistake :P

**Comment by chaithyagr** [comment]
I added that we could have additional support for learning a sampling pattern in 2D with straight line readouts. 

Perhaps we can do it while supporting stackNUFFT on torch? @paquiteau is this possible? Can we pass the backprop gradients through the stacked NUFFT from the internal plane NUFFT?

**Comment by chaithyagr** [comment]
On second thoughts, we would need 176 2D NUFFT then `num_samples` number of FFTs, I think a simple straightforward 3D NUFFT can be easier. @paquiteau to comment. I think in your usecase for fMRI it is more useful.

**Comment by paquiteau** [comment]
after rethinking about it, stacked nufft does not have autodiff wrt to trajectory. It does support the autodiff wrt to data however.

**Comment by chaithyagr** [comment]
I think the only issue left here is to do is based on deep inverse. I feel we should open a different issue for writing example to do PnP reconstructions.
